Sources/BraveWallet/Crypto/Stores/NetworkStore.swift

```diff
@@ -57,61 +73,81 @@ public class NetworkStore: ObservableObject {
+    self.selectedChainIdForOrigin = chainForOrigin.chainId
+    // update `isSwapSupported` for Buy/Send/Swap panel
+    self.isSwapSupported = await swapService.isSwapSupported(chain.chainId)
   }
 
-  private func updateChainList() {
-    Task { @MainActor in // fetch all networks for all coin types
-      self.allChains = await rpcService.allNetworksForSupportedCoins()
-      
-      let customChainIds = await rpcService.customNetworks(.eth) // only support Ethereum custom chains
-      self.customChains = allChains.filter { customChainIds.contains($0.id) }
-    }
+  @MainActor private func updateChainList() async {
+    // fetch all networks for all coin types
+    self.allChains = await rpcService.allNetworksForSupportedCoins()
+    
+    let customChainIds = await rpcService.customNetworks(.eth) // only support Ethereum custom chains
+    self.customChains = allChains.filter { customChainIds.contains($0.id) }
   }
   
   func isCustomChain(_ network: BraveWallet.NetworkInfo) -> Bool {
     customChains.contains(where: { $0.coin == network.coin && $0.chainId == network.chainId })
   }
   
-  @MainActor @discardableResult func setSelectedChain(_ network: BraveWallet.NetworkInfo) async -> SetSelectedChainError? {
+  @MainActor @discardableResult func setSelectedChain(
+    _ network: BraveWallet.NetworkInfo,
+    isForOrigin: Bool
+  ) async -> SetSelectedChainError? {
     let keyringId = network.coin.keyringId
     let keyringInfo = await keyringService.keyringInfo(keyringId)
     if keyringInfo.accountInfos.isEmpty {
       // Need to prompt user to create new account via alert
       return .selectedChainHasNoAccounts
     } else {
       let selectedCoin = await walletService.selectedCoin()
-      if self.selectedChainId != network.chainId {
-        self.selectedChainId = network.chainId
+      if isForOrigin {
+        if self.selectedChainIdForOrigin != network.chainId {
+          self.selectedChainIdForOrigin = network.chainId
+        }
+      } else {
+        if self.defaultSelectedChainId != network.chainId {
+          self.defaultSelectedChainId = network.chainId
+        }
       }
       if selectedCoin != network.coin {
-        let success = await rpcService.setNetwork(network.chainId, coin: network.coin, origin: nil)
+        let success = await rpcService.setNetwork(network.chainId, coin: network.coin, origin: isForOrigin ? origin : nil)
         return success ? nil : .unknown
       } else {
-        let rpcServiceNetwork = await rpcService.network(network.coin, origin: nil)
+        let rpcServiceNetwork = await rpcService.network(network.coin, origin: isForOrigin ? origin : nil)
         guard rpcServiceNetwork.chainId != network.chainId else {
-          self.isSwapSupported = await swapService.isSwapSupported(network.chainId)
+          if !isForOrigin { // `isSwapSupported` is for the `defaultSelectedChain`
+            self.isSwapSupported = await swapService.isSwapSupported(network.chainId)
+          }
```

Comment:
`isSwapSupported` needs to be updated for dapps too i think. user clicks the buy/send/swap button from wallet panel

Reply:
Buy/Send/Swap uses the `defaultSelectedChain` even when opened from the panel, so I believe `isSwapSupported` should be tied to that network.